src/main/java/strategy/PurchaseOperation.java

```
@@ -0,0 +1,18 @@
+package strategy;
+
+import db.Storage;
+import model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int current = Storage.getQuantity(fruit);
+        int newBalance = current - transaction.getQuantity();
+        if (newBalance < 0) {
+            throw new RuntimeException("Negative balance for fruit: " + fruit);
+        }
+        Storage.setQuantity(fruit, newBalance);
```

After computing the new balance you call `Storage.setQuantity(...)` without guarding or wrapping potential exceptions with context. The checklist requires: "Do not ignore exceptions; rethrow as RuntimeException with informative messages and original cause." If `Storage` can throw, catch exceptions, wrap them in a RuntimeException that includes transaction details and the original cause, and rethrow so failures are informative.